ykrt/src/mt.rs

```diff
@@ -235,6 +236,7 @@ impl MTInner {
 }
 
 type CompilingTrace<I> = Mutex<Option<Box<CompiledTrace<I>>>>;
+struct ThreadIdInner(u8);
```

Comment:
I think this can be a unit struct. The `Arc` will have to allocate memory any way to fit the strong and weak reference count.

Reply:
I remember reading that there are some `malloc`s somewhere that don't actually allocate memory if you ask for 0 bytes, so I'm a bit nervous about that. i also half-wondered if I should use a `usize` here to force word-level alignment, but that might be unduly paranoid of me!

Reply:
The `Arc` itself will allocate a memory locatikn big enough to fit the pointer sized strong and weak reference counters before the actual data. https://github.com/rust-lang/rust/blob/2ba7ca2bbbff6cd424aebc654308febc00b9497a/library/alloc/src/sync.rs#L301

Reply:
Good point! Fixed in https://github.com/softdevteam/yk/pull/195/commits/3d21b29494100c9b7a62c71c1cc59c3fe5685681.